src/models/stores/serial.ts

```diff
@@ -62,7 +63,15 @@ export class SerialDevice {
   }
 
   public async handleStream(channels: Array<NodeChannelInfo>){
+    // open port
     await this.port?.open({ baudRate: 9600 }).catch((e: any) => console.error(e));
```

Comment:
Rather than using optional chaining throughout this function (`this.port?....`) we should probably just test for a `null` port at the top and bail. Passing `null` to the `pipeTo` method probably isn't supported, for instance.